authserver/token.go

```diff
@@ -151,19 +140,18 @@ func (t *TokenResource) Handle(tokenRequest TokenRequest) (*TokenResponse, *RFC6
 
 	jti := base64.RawURLEncoding.EncodeToString(rawJTI)
 
-	accessTokenEpiresIn := jwt.NewNumericDate(now.Add(10 * time.Minute))
+	accessTokenEpiresIn := now.Add(10 * time.Minute)
 
-	claims := jwt.Claims{
-		Issuer:    t.origin,
-		Subject:   subject,
-		Audience:  []string{t.origin},
-		Expiry:    accessTokenEpiresIn,
-		NotBefore: jwt.NewNumericDate(now),
-		IssuedAt:  jwt.NewNumericDate(now),
-		ID:        jti,
+	claims := jwt.ClaimSet{
+		Issuer:   t.origin,
+		Subject:  subject,
+		Audience: []string{t.origin},
+		Expiry:   accessTokenEpiresIn.Unix(),
+		IssuedAt: now.Unix(),
+		JwtId:    jti,
 	}
 
-	accessToken, err := jwt.Signed(signer).Claims(claims).CompactSerialize()
+	accessToken, err := jwt.EncodeAndSign(&claims, t.privateKeyId, t.privateKey)
```

Comment:
Yeah, I notice this kind of thing a lot in Go code in the wild, and I think you’re right that it comes from C, and other similar languages where heap v. stack allocation is a Big Deal™, and you want to avoid heap allocations when you can. Meanwhile, Go DGAF if pointer or value, and puts it wherever it thinks it needs to go.